tools/lab_dispatch/generic_source.yaml

```diff
@@ -0,0 +1,78 @@
+job_queue: $INPUT_QUEUE
+global_timeout: 3600
+output_timeout: 1800
+provision_data:
+  $INPUT_DATA_SOURCE
+test_data:
+  attachments:
+    - local: "tools/lab_dispatch/resources/manifest.conf"
+      agent: "manifest.conf"
+    - local: "tools/lab_dispatch/resources/input.conf"
+      agent: "input.conf"
+    - local: "tools/lab_dispatch/resources/service_file.service"
+      agent: "service_file.service"
+    - local: "tools/lab_dispatch/build_install_deb.py"
+      agent: "build_install_deb.py"
+  test_cmds: |
+    #!/usr/bin/env bash
+
+    set -x
+
+    # input arguments
+    CHECKBOX_REVISION=$INPUT_CHECKBOX_REVISION
+    RESOURCES_PATH="attachments/test"
+
+    TOOLS_PATH=tools
+
+    # retrieve all scripts/tools necessary from a repo
+    curl -Ls -o install_tools.sh https://raw.githubusercontent.com/canonical/hwcert-jenkins-tools/main/install_tools.sh
+    source install_tools.sh $TOOLS_PATH
+
+    # ensure device is available before continuing
+    wait_for_ssh
+
+    _run sudo add-apt-repository ppa:checkbox-dev/edge
+    _run install_packages git python3 python3-pip dpkg-dev
+
+    wait_for_ssh
+
+    _put $RESOURCES_PATH/build_install_deb.py :
+    _run git clone https://github.com/canonical/checkbox.git
+    _run git -C checkbox checkout $CHECKBOX_REVISION
+    _run python3 build_install_deb.py --clean checkbox/checkbox-ng \
+        checkbox/checkbox-support checkbox/providers/resource \
+        checkbox/providers/base checkbox/providers/sru
+    _run sudo systemctl restart checkbox-ng.service
```

Comment:
If you like, you can omit `.service`.

Reply:
I always include the extension out of habit